Write a document around Cloud data storage and controls (#5425)
cypress-io/cypress-documentation: docs/guides/cloud/data-storage-and-masking.mdx

---
title: Data Storage and Controls
sidebar_position: 60
---

:::info

## <Icon name="graduation-cap" /> What you'll learn

- What data Cypress Cloud stores

> - What controls exist to control data sent to and viewed in Cypress Cloud

Reviewer: ```suggestion
- What controls exist to manage data sent to and viewed in Cypress Cloud
```
minor: change up a word to avoid "what controls... control"

Author: Addressed in [`06a38a9` (#5425)](https://github.com/cypress-io/cypress-documentation/pull/5425/commits/06a38a963c8df47c79ad35fcafc4a29b616249a1)

:::

## Data Storage

:::caution

Please refer to the [Cypress Cloud Terms of Use](https://cloud.cypress.io/terms-of-use) and our [Security & Compliance](https://www.cypress.io/security/) guide for more specifics around reasonable use of Cypress Cloud, data storage and security at Cypress.

:::

When you run Cypress via `cypress run` and pass the `--record` flag, the following data is stored in Cypress Cloud for every run:

- Standard output in the terminal
- Test results
- Test definitions
- Cypress configuration (minus Cypress environment variables)
- Screenshots
- Videos
- OS environment variables related to CI and git information

> When [Test Replay](#Test-Replay) capturing is enabled, the additional data outlined below will be sent to Cypress Cloud. [See our guide on disabling Test Replay.](/guides/cloud/debugging/test-replay#Opt-out-of-Test-Replay)

Reviewer 2: The test replay hash still needs somewhere to link to

- The rendered DOM and CSS styles for the application under test
- Cypress commands and events represented in the Command Log
- Network traffic within your application under test
- Browser console logs

Cypress Cloud does not capture anything related to the code of your application under test or the code within any repositories that are associated with your project.

## Cypress Cloud data controls

We understand that controlling what data is sent to Cypress Cloud is important to our users. Cypress provides a number of ways to control what data is sent to Cypress Cloud.

If you'd like more granular control over what data is sent to Cypress Cloud, please comment in this discussion: TODO:ADD A LINK TO DISCUSSION or [contact us](https://www.cypress.io/contact/) to discuss your needs.

### Project access controls

Cypress Cloud offers controls around the visibility of a project. You can set a project to be public or private. [See project access control settings.](/guides/cloud/account-management/projects#Public-vs-Private)

> Setting a project to private means that users invited to the organization will have visibility into the project, its runs, and any data captured by Cypress.

Reviewer: ```suggestion
Setting a project to private means that only the users explicitly invited to the organization will have visibility into the project, its runs, and any data captured by Cypress.
```
Contemplating whether the extra emphasis here helps to drive home the point. Take it if you like it.

Author: Addressed in [`eb566cc` (#5425)](https://github.com/cypress-io/cypress-documentation/pull/5425/commits/eb566cca4cf3b6e79c8aba1084cb8be1a495c69f)

  cy.visit('/admin')
  cy.screenshot({
    blackout: ['.user-name'],
  })
})
```

#### Remove Cypress Command Log from screenshots

Cypress offers an option to hide the Cypress Command Log from screenshots so that no information from that area will be captured in screenshots. [See `cy.screenshot` docs](/api/commands/screenshot#Arguments).

Add a control to your [support file](/guides/references/configuration#Testing-Type-Specific-Options) to set the default capture options for all screenshots.

```js
before(() => {
  Cypress.Screenshot.defaults({
    capture: 'viewport', // or fullPage
  })
})
```

Add a control for a specific screenshot to only capture the application under test.

```js
it('sign in page', () => {
  cy.visit('/sign-in')
  cy.screenshot({
    capture: 'viewport', // or fullPage
  })
})
```

### Video controls

#### Disable video upload

You can delete videos that are recorded when you are passing `video: true` by following [this guide to control what is uploaded to Cypress Cloud](/guides/guides/screenshots-and-videos#Control-which-videos-to-keep-and-upload-to-Cypress-Cloud).

### Cypress Command Log controls

#### Turn off Cypress Command Log rendering

> You can disable showing the entirety of the Cypress Command Log as well as the Application under Test (AUT) controls in your tests to prevent all test content from being shown and recorded in any screenshot or video by passing `--no-runner-ui` via the CLI while running Cypress tests. [See docs.](/guides/guides/command-line#Options). This means that the AUT will be the only thing that is rendered while running tests, so the only thing that is captured in screenshots and videos.

Reviewer 2: Ended up with a double period here. Also one more more descriptive text link. 

```suggestion
You can disable showing the entirety of the Cypress Command Log as well as the Application under Test (AUT) controls in your tests to prevent all test content from being shown and recorded in any screenshot or video by passing `--no-runner-ui` via the CLI while running Cypress tests [See Cypress Command Line options.](/guides/guides/command-line#Options). This means that the AUT will be the only thing that is rendered while running tests, so the only thing that is captured in screenshots and videos.
```

---
title: Data Storage and Controls
sidebar_position: 60
sidebar_label: Data Storage and Controls

> ---

Reviewer 2: Shouldn't need a sidebar label value here since the page title is the same and will dictate the sidebar label.

Author: Makes sense.

Author: Addressed om [`9838d09` (#5425)](https://github.com/cypress-io/cypress-documentation/pull/5425/commits/9838d0989532a0989f7e44c2bdd9b3346fc37ac4)

:::info

## <Icon name="graduation-cap" /> What you'll learn

> - What data does Cypress Cloud store?

Reviewer 2: Tiny thing, but I don't believe we frame these points as questions in any other page.

- What controls exist to control data sent to and viewed in Cypress Cloud?

:::

## Data Storage

:::caution

> Please refer to the [Cypress Cloud Terms of Use](https://cloud.cypress.io/terms-of-use) and our [Security & Compliance](https://www.cypress.io/security/) guide for more specifics around reasonable use of Cypress Cloud, data storage and security at Cypress.

Reviewer 2: What do we mean when we say "reasonable use"? I think we may be able to reword this a bit to streamline it, but I wanted to clarify the points.

:::

> When you run Cypress via `cypress run` passing the `--record` flag the following data is stored in Cypress Cloud for every run:

Reviewer 2: ```suggestion
When you run Cypress via `cypress run` and pass the `--record` flag, the following data is stored in Cypress Cloud for every run:
```

- Standard output in the terminal
- Test results
- Test definitions
- Cypress configuration (minus Cypress environment variables)
- Screenshots
- Videos
- OS environment variables related to CI and git information

> When [Test Replay](#Test-Replay) capturing is enabled, the additional data outlined below will be sent to the Cloud. You can disable capturing: TODO link this data.

Reviewer 2: Should this be linked to the main TR product page?

Reviewer 2: also the disable portion will link to `/guides/cloud/debugging/test-replay#Opt-out-of-Test-Replay`

Reviewer 2: We refer to what's stored in Cypress Cloud in the first list above, but just call it Cloud here. Could go either way depending on the context, but we should make them consistent.

Author: Address Cypress Cloud part in [`0b22fd0` (#5425)](https://github.com/cypress-io/cypress-documentation/pull/5425/commits/0b22fd0a871b2d32d1f79d79fcb4b2fc51ee4d08)

Yah, a lot of this is hard to do without having the Test Replay merged into the v13 branch. I'll have to update the links after.

- The rendered DOM and CSS styles for the application under test
- Cypress commands and events represented in the Command Log
- Network traffic within your application under test
- Browser console logs

> Cypress Cloud does not capture anything related to the code of your application under test or the code within any associated repositories that are associated with your project.

Reviewer 3: ```suggestion
Cypress Cloud does not capture anything related to the code of your application under test or the code within any repositories that are associated with your project.
```

> ## Cypress Cloud data cntrols

Reviewer 2: ```suggestion
## Cypress Cloud data controls
```

We understand that controlling what data is sent to Cypress Cloud is important to our users. Cypress provides a number of ways to control what data is sent to Cypress Cloud.

If you'd like more granular control over what data is sent to Cypress Cloud, please comment in this discussion: TODO:ADD A LINK TO DISCUSSION or [contact us](https://www.cypress.io/contact/) to discuss your needs.

### Project access controls

> Cypress Cloud offers controls around the visibility of a project. You can set a project to be public or private. [See docs.](/guides/cloud/account-management/projects#Public-vs-Private)

Reviewer 2: We can be more granular with links for clarity and even for SEO purposes. 

```suggestion
Cypress Cloud offers controls around the visibility of a project. You can set a project to be public or private. [See project access control settings.](/guides/cloud/account-management/projects#Public-vs-Private)
```

Setting a project to private means that users invited to the organization will have visibility into the project, its runs, and any data captured by Cypress.

### Test Replay controls

#### Disable Test Replay

> You can disable capturing data in Test Replay per project by following this guide: TODO: Test Replay guide link. This means Test Replay will not be available for viewing for the project. This will prevent Cypress Cloud from capturing the following data:

Reviewer 2: `/guides/cloud/debugging/test-replay#Opt-out-of-Test-Replay`

Author: Will address when test Replay doc is merged upstream

> You can disable capturing data in Test Replay per project by following this guide: TODO: Test Replay guide link. This means Test Replay will not be available for viewing for the project. This will prevent Cypress Cloud from capturing the following data:

Reviewer 2: ```suggestion
You can disable capturing data in Test Replay per project by following this guide: TODO: Test Replay guide link. This means Test Replay will not be available for viewing or debugging in the project. This will prevent Cypress Cloud from capturing the following data:
```

- The rendered DOM and CSS styles for the application under test
- Cypress commands and events represented in the Command Log
- Network traffic within your application under test
- Browser console logs

### Screenshot controls

#### Blackout elements in screenshots

> Cypress offers an option to black out (mask) an array of elements from screenshots. [See docs.](/api/commands/screenshot#Arguments) This allows you to black out specific elements on screenshot so that they will not be visible in the locally saved screenshots or to any users that have access to the run in Cypress Cloud.

Reviewer 2: ```suggestion
Cypress offers an option to black out (mask) an array of elements from screenshots. [See screenshot API docs.](/api/commands/screenshot#Arguments) This allows you to black out specific elements on screenshot so that they will not be visible in the locally saved screenshots or to any users that have access to the run in Cypress Cloud.
```

> Cypress offers an option to black out (mask) an array of elements from screenshots. [See docs.](/api/commands/screenshot#Arguments) This allows you to black out specific elements on screenshot so that they will not be visible in the locally saved screenshots or to any users that have access to the run in Cypress Cloud.

Author: Addressed in [`8fedac7` (#5425)](https://github.com/cypress-io/cypress-documentation/pull/5425/commits/8fedac7c7ffdb865dd44398a67a52fff00dee690)

> Add a control to your `cypress/support/index.js` file to set the default blackout options for all screenshots.

Reviewer 2: we may want to link out to [this config table](https://docs.cypress.io/guides/references/configuration#Testing-Type-Specific-Options), since the path to the support file could be different here. Either way it wouldn't be `cypress/support/index.js`

Author: Addressed in [`dea094a` (#5425)](https://github.com/cypress-io/cypress-documentation/pull/5425/commits/dea094a2c942767dce51b9829db4fa67e48b9933)

```js
before(() => {
  Cypress.Screenshot.defaults({
    // blackout all elements with 'mask-cy' attribute
    blackout: ['mask-cy'],
  })
})
```

Add a control for a specific screenshot to blackout specific elements.

```js
it('admin page loads', () => {
  cy.visit('/admin')
  cy.screenshot({
    blackout: ['.user-name'],
  })
})
```

#### Remove Cypress Command Log from screenshots

> Cypress offers an option to hide the Cypress Command Log from screenshots so that no information from that area will captured in screenshots. [See docs](/api/commands/screenshot#Arguments).

Reviewer 2: same here for the link text & support path

Author: Addressed in [`afdd515` (#5425)](https://github.com/cypress-io/cypress-documentation/pull/5425/commits/afdd515eab51c8a1a5c402603b2ee5afd0640d69)

Add a control to your `cypress/support/index.js` file to set the default capture options for all screenshots.

```js
before(() => {
  Cypress.Screenshot.defaults({
    capture: 'viewport', // or fullPage
  })
})
```

Add a control for a specific screenshot to only capture the application under test.

```js
it('sign in page', () => {
  cy.visit('/sign-in')
  cy.screenshot({
    capture: 'viewport', // or fullPage
  })
})
```

### Video controls

#### Disable video upload

You can delete videos that are recorded when you are passing `video: true` by following [this guide](guides/guides/screenshots-and-videos#Control-which-videos-to-keep-and-upload-to-Cypress-Cloud). This prevents them from being uploaded to Cypress Cloud.

### Cypress Command Log controls

#### Turn off Cypress Command Log rendering

> You can disable showing the entirety of the Cypress Command Log as well as the AUT controls in your tests to prevent all test content from being shown and recorded in any screenshot or video by passing `--no-runner-ui` via the CLI while running Cypress tests. [See docs.](/guides/guides/command-line#Options). This means that the application under test will be the only thing that is rendered while running tests, so the only thing that is captured in screenshots and videos.

Reviewer 2: [See Cypress command line docs.]

> Note that when `--no-runner-ui` is set, data from the Cypress Command Log will still be sent to Cypress Cloud to be generated and displayed in Test Replay if Test Replay is enabled. See [Disable Test Replay](#Disable-Test-Replay) for more information.

Reviewer 3: ```suggestion
Note that when `--no-runner-ui` is set, data from the Cypress Command Log will still be sent to Cypress Cloud to be generated and displayed in Test Replay if it is enabled. See [Disable Test Replay](#Disable-Test-Replay) for more information.
```

#### Mask data in the Cypress Command Log

There are multiple ways to mask data in the Cypress Command Log. Note that when these strategies are used, data within test definitions will still be sent to Cypress Cloud.

##### `log: false` for Cypress commands

> You can pass `log: false` as an option to Cypress specific commands in your tests to prevent the data from being shown visibly in the Cypress Command Log and being shown in any captured screenshots or videos.

Reviewer 2: ```suggestion
You can pass `log: false` as an option to Cypress-specific commands in your tests to prevent the data from being shown visibly in the Cypress Command Log and being shown in any captured screenshots or videos.
```

Reviewer 2: It may be useful to show a basic user/pw `.type()` scenario here in a code block to reiterate the `log: false` example IRL

Reviewer 2: nm. I see we're showing this down below

##### Disable logging of XHR/Fetch requests

By default, Cypress logs all requests that match any [`cy.intercept()`](/api/commands/intercept) in the Cypress Command Log, as well as all XMLHttpRequests and fetch requests. You can use `cy.intercept()` to disable these logs by passing `{ log: false }` in the second parameter. [See docs.](/api/commands/intercept#Disabling-logs-for-a-request)

```js
before(() => {
  // disable Cypress's default behavior of logging all XMLHttpRequests and fetches
  cy.intercept({ resourceType: /xhr|fetch/ }, { log: false })
})
```

##### Mask Cypress commands in the Cypress Command Log

> Users can mask passwords in the Cypress Command Log by using a custom command like the code below. [See docs.](/api/cypress-api/custom-commands#Overwrite-type-command)

Reviewer 2: ```suggestion
Users can mask passwords in the Cypress Command Log by using a custom command like the code below. [See our guide on masking the contents of the `.type()` command.](/api/cypress-api/custom-commands#Overwrite-type-command)
```

- Test results
- Test definitions
- Cypress configuration (minus Cypress environment variables)
- Screenshots
- Videos
- OS environment variables related to CI and git information

When [Test Replay](/guides/cloud/debugging/test-replay) capturing is enabled, the additional data outlined below will be sent to Cypress Cloud. [See our guide on disabling Test Replay.](/guides/cloud/debugging/test-replay#Opt-out-of-Test-Replay)

- The rendered DOM and CSS styles for the application under test
- Cypress commands and events represented in the Command Log
- Network traffic within your application under test
- Browser console logs

Cypress Cloud does not capture anything related to the code of your application under test or the code within any repositories that are associated with your project.

You, the customer, own and are responsible for your test content. We trust that customers will make decisions about the appropriateness of data to use in testing and will avoid use of PII, PHI, or other types of protected information. Cypress makes commercially reasonable efforts to protect confidentiality of information that is provided to Cypress Cloud.

## Cypress Cloud data controls

> We understand that controlling what data is sent to Cypress Cloud is important to our users. Cypress provides a number of ways to control what data is sent to Cypress Cloud.

Reviewer 3: ```suggestion
We understand that controlling what data is sent to Cypress Cloud is important to our users. Cypress provides a number of ways to control this data transfer.
```

Attempting to avoid repetition.

  cy.visit('/admin')
  cy.screenshot({
    blackout: ['.user-name'],
  })
})
```

#### Remove Cypress Command Log from screenshots

Cypress offers an option to hide the Cypress Command Log from screenshots so that no information from that area will be captured in screenshots. [See `cy.screenshot` docs](/api/commands/screenshot#Arguments).

Add a control to your [support file](/guides/references/configuration#Testing-Type-Specific-Options) to set the default capture options for all screenshots.

```js
before(() => {
  Cypress.Screenshot.defaults({
    capture: 'viewport', // or fullPage
  })
})
```

Add a control for a specific screenshot to only capture the application under test.

```js
it('sign in page', () => {
  cy.visit('/sign-in')
  cy.screenshot({
    capture: 'viewport', // or fullPage
  })
})
```

### Video controls

#### Disable video upload

You can delete videos that are recorded when you are passing `video: true` by following [this guide to control what is uploaded to Cypress Cloud](/guides/guides/screenshots-and-videos#Control-which-videos-to-keep-and-upload-to-Cypress-Cloud).

### Cypress Command Log controls

#### Turn off Cypress Command Log rendering

> You can disable showing the entirety of the Cypress Command Log as well as the Application under Test (AUT) controls in your tests to prevent all test content from being shown and recorded in any screenshot or video by passing `--no-runner-ui` via the CLI while running Cypress tests [See Cypress Command Line options.](/guides/guides/command-line#Options). This means that the AUT will be the only thing that is rendered while running tests, so the only thing that is captured in screenshots and videos.

Reviewer 3: ```suggestion
You can disable showing the entirety of the Cypress Command Log as well as the Application under Test (AUT) controls in your tests to prevent all test content from being shown and recorded in any screenshot or video by passing `--no-runner-ui` via the CLI while running Cypress tests. See [Cypress Command Line options](/guides/guides/command-line#Options). This means that the AUT will be the only thing that is rendered while running tests, so the only thing that is captured in screenshots and videos.
```

##### `log: false` for Cypress commands

You can pass `log: false` as an option to Cypress-specific commands in your tests to prevent the data from being shown visibly in the Cypress Command Log and being shown in any captured screenshots or videos.

##### Disable logging of XHR/Fetch requests

By default, Cypress logs all requests that match any [`cy.intercept()`](/api/commands/intercept) in the Cypress Command Log, as well as all XMLHttpRequests and fetch requests. You can use `cy.intercept()` to disable these logs by passing `{ log: false }` in the second parameter. [See `cy.intercept()` docs.](/api/commands/intercept#Disabling-logs-for-a-request)

```js
before(() => {
  // disable Cypress's default behavior of logging all XMLHttpRequests and fetches
  cy.intercept({ resourceType: /xhr|fetch/ }, { log: false })
})
```

##### Mask Cypress commands in the Cypress Command Log

Users can mask passwords in the Cypress Command Log by using a custom command like the code below. [See our guide on masking the contents of the `.type()` command.](/api/cypress-api/custom-commands#Overwrite-type-command)

> <DocsImage src="/img/api/custom-commands/custom-command-type-masked-password.png" />

Reviewer 3: This screenshot is out of date. It should not show `blacklisted-host`, since it is now `blocked-host`

![image](https://github.com/cypress-io/cypress-documentation/assets/66998419/7a74a6b4-b4bd-4a58-9ad9-b0276f9f0cf2)

Author: hmm, yah took this from the blockHosts page.

Author: Updated in [`7f241b3` (#5425)](https://github.com/cypress-io/cypress-documentation/pull/5425/commits/7f241b3381fa9a5f4c0af3241453a6fc5ef27d72)

- Videos
- OS environment variables related to CI and git information

> When [Test Replay](#Test-Replay) capturing is enabled, the additional data outlined below will be sent to the Cypress Cloud. You can disable capturing: TODO link this data.

Reviewer 2: probably just, Cypress Cloud. can loose "the".

Author: Addressed in [`5281bbf` (#5425)](https://github.com/cypress-io/cypress-documentation/pull/5425/commits/5281bbf6bae773d499bdd69b81999f88db93b871)
